packages/core/src/storage/StorageService.ts

```diff
@@ -12,7 +12,13 @@ interface AdvancedQuery<T extends BaseRecord> {
   $not?: Query<T>
 }
 
-export type Query<T extends BaseRecord<any, any, any>> = AdvancedQuery<T> | SimpleQuery<T>
+interface PaginationQuery {
```

Comment:
I agree with @TimoGlastra in that maybe using an extra query options parameters would be cleaner. What about something like the following?

```ts
export type QueryOptions = {
  limit?: number
  offset?: number
}

findByQuery(agentContext: AgentContext, recordClass: BaseRecordConstructor<T>, query: Query<T>, queryOptions?: QueryOptions): Promise<T[]>
```